mango/clients/rest_client.py

```diff
@@ -0,0 +1,85 @@
+import time
+import warnings
+from abc import ABC, abstractmethod
+from typing import Union, Literal, Type, List
+
+import requests
+from pydantic import BaseModel
+
+
+class RestClient(ABC):
```

Comment:
I would call it `RESTClient`. Could you look up how we have don the `cornflow-client` class and see if there is anything there that could be reused?

I like the part where for typical calls it will automatically check for the status codes and will give you back the data, but all calls can be done to get the status code and the response object.

Reply:
I'll take a look. Can you explain the second paragraph?

Reply:
Name changed.

Need to take a look into cornflow-client